### PR TITLE
[ML] Support queue settings property for command jobs

### PR DESCRIPTION
**Pull request by vivijay91**
# Description

Added QueueSettings schema and entity. Updated command job schema to use queue settings.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/queue_settings.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# ---------------------------------------------------------
+
+# pylint: disable=unused-argument,no-self-use
+
+from marshmallow import fields, post_load
+
+from azure.ai.ml._schema.core.schema_meta import PatchedSchemaMeta
+
+
+class QueueSettingsSchema(metaclass=PatchedSchemaMeta):
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/queue_settings.py:12]: @vivijay91 I am working on a [PR to support queue_settings](https://github.com/Azure/azure-sdk-for-python/pull/29122). Looks like we need to join efforts here. Thanks!

File: sdk/ml/azure-ai-ml/tests/dsl/unittests/test_command_builder.py [hunk 2]
```diff
@@ -680,6 +680,27 @@ def test_resources_from_dict(self, test_command_params):
         rest_dict = command_node._to_rest_object()
         assert rest_dict["resources"] == {"instance_type": "STANDARD_D2"}
 
+    def test_queue_settings(self, test_command_params):
+        expected_queue_settings = {"job_tier": "standard"}
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/tests/dsl/unittests/test_command_builder.py:684]: `{"job_tier": "standard", "priority":"medium"}`
Can we please include `priorty` as part of the test?

File: sdk/ml/azure-ai-ml/tests/command_job/unittests/test_command_job_entity.py [hunk 3]
```diff
@@ -134,6 +135,7 @@ def test_command_job_builder_serialization(self) -> None:
             instance_type="STANDARD_BLA",
             timeout=300,
             code="./",
+            queue_settings=QueueSettings(job_tier="standard")
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/tests/command_job/unittests/test_command_job_entity.py:138]: Can we please include `priorty="medium"` as part of the test?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/base_job.py [hunk 4]
```diff
@@ -59,7 +59,7 @@ class BaseJobSchema(ResourceSchema):
             )
         },
     )
-    compute = ComputeField(required=True)
+    compute = ComputeField(required=False)
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/base_job.py:62]: Just curious, is ComputeField optional now?

Comment (vivijay91) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/base_job.py:62]: Yes, serverless compute (which is in preview and what we need queue settings for) will be used when compute is not specified

Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/base_job.py:62]: Got it. If both compute and queue_settings are none, should we assign defaults for tier & priority or will it be taken care by the backend?

Comment (vivijay91) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/base_job.py:62]: It will be taken care of by backend

File: sdk/ml/azure-ai-ml/tests/dsl/unittests/test_command_builder.py (hunk 2 above)
Comment (vivijay91) [reply sdk/ml/azure-ai-ml/tests/dsl/unittests/test_command_builder.py]: Done

File: sdk/ml/azure-ai-ml/tests/command_job/unittests/test_command_job_entity.py (hunk 3 above)
Comment (vivijay91) [reply sdk/ml/azure-ai-ml/tests/command_job/unittests/test_command_job_entity.py]: Done

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py [hunk 5]
```diff
@@ -40,3 +41,4 @@ class ParameterizedCommandSchema(PathAwareSchema):
     )
     resources = NestedField(JobResourceConfigurationSchema)
     distribution = DistributionField()
+    queue_settings = NestedField(QueueSettingsSchema)
```
Comment (zhengfeiwang) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py:44]: Also need to do this for `CommandSchema` in `_schema/pipeline/component_job.py`

**Comment by zhengfeiwang** [comment]
For `CommandJob._to_node` in `entities/_job/command_job.py`, need to pass in `queue_settings` for the returning `Command`.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py (hunk 5 above)
Comment (vivijay91) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py:44]: `CommandSchema` in `_schema/pipeline/component_job.py` inherits from `ParameterizedCommandSchema` which already has `queue_settings` defined. Is it still expected to be added in that file?

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py:44]: I'm also confused for this, but during my test on your branch, the `queue_settings` keeps `None`; after I add it in `CommandSchema`, everything goes fine. Not sure if it is something magic inner `marshmallow`.

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py:44]: Also, you can merge this PR without the above change. I will open a PR for Pipelines after this merged.

Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py:44]: Can you wrap this field in `ExperimentalField()`? Thx.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py [hunk 6]
```diff
@@ -4,7 +4,7 @@
 # pylint: disable=protected-access
 from typing import Any, Iterable
 
-from azure.ai.ml._restclient.v2022_10_01 import AzureMachineLearningWorkspaces as ServiceClient102022
+from azure.ai.ml._restclient.v2023_02_01_preview import AzureMachineLearningWorkspaces as ServiceClient022023Preview
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py:7]: Why did you need to change this?

**Review by luigiw — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py (hunk 6 above)
Comment (vivijay91) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py:7]: schedule command e2e job operations were failing without this change since we're trying to read queue settings from the rest object
